cmd/neofs-node/container.go

```diff
@@ -79,11 +83,135 @@ func initContainerService(c *cfg) {
+	}
+	durrSec, err := c.nCli.EpochDuration()
+	if err != nil {
+		fatalOnErr(fmt.Errorf("failed to get epoch duration to initialize container service: %w", err))
+	}
+	durr := time.Duration(durrSec) * time.Second
+
+	var (
+		indexInNM = -1
+		nmLen     = len(nm.Nodes())
+	)
+	for i, node := range nm.Nodes() {
+		if bytes.Equal(node.PublicKey(), c.binPublicKey) {
+			indexInNM = i
+			break
+		}
+	}
+	if indexInNM == -1 {
+		indexInNM = nmLen
+	}
+
+	var (
+		reportStep   time.Duration
+		stepsInEpoch uint32
+	)
+	if nmLen != 0 {
+		reportStep = min(time.Second, durr/maxReportsPerEpoch/time.Duration(nmLen))
+		stepsInEpoch = uint32(durr / reportStep)
+	}
+
+	var (
+		ticks      timer.EpochTicks
+		reportTick = reportHandler(c, l)
+	)
+	for i := range maxReportsPerEpoch {
+		var mul, div uint32
+		if stepsInEpoch == 0 {
+			mul = uint32(i)
+			div = maxReportsPerEpoch
+		} else {
+			mul = uint32(i)*(stepsInEpoch/maxReportsPerEpoch) + uint32(indexInNM)
+			div = stepsInEpoch
+		}
+
+		l.Debug("add space load reporter", zap.Uint32("multiplicator", mul), zap.Uint32("divisor", div))
+
+		ticks.DeltaTicks = append(ticks.DeltaTicks, timer.SubEpochTick{
+			Tick:     reportTick,
+			EpochMul: mul,
+			EpochDiv: div,
+		})
+	}
+
+	c.cfgMorph.epochTimers = timer.NewTimers(ticks)
+	addNewEpochAsyncNotificationHandler(c, func(ev event.Event) {
+		txHash := ev.(netmap.NewEpoch).TxHash()
+		txHeight, err := c.nCli.Morph().TxHeight(txHash)
+		if err != nil {
+			l.Warn("can't get transaction height", zap.String("hash", txHash.StringLE()), zap.Error(err))
+			return
+		}
+		epochDuration, err := c.nCli.EpochDuration()
+		if err != nil {
+			l.Warn("failed to get the epoch duration to reset epoch timer", zap.Error(err))
+			return
+		}
+		lastTick := txHeight
+		if lastTick == 0 {
+			lastTick, err = c.nCli.LastEpochBlock()
+			if err != nil {
+				l.Warn("failed to get the last epoch block number to reset epoch timer", zap.Error(err))
+				return
+			}
+		}
+		lastTickH, err := c.cli.GetBlockHeader(lastTick)
+		if err != nil {
+			l.Warn("failed to get the last epoch block to reset epoch timer", zap.Error(err))
+			return
+		}
+		height, err := c.cli.GetBlockCount()
+		if err != nil {
+			l.Warn("failed to get the current height to reset epoch timer", zap.Error(err))
+			return
+		}
+		currH, err := c.cli.GetBlockHeader(height - 1)
+		if err != nil {
+			l.Warn("failed to get the current block", zap.Error(err))
+			return
+		}
+
+		const msInS = 1000
+		c.cfgMorph.epochTimers.Reset(lastTickH.Timestamp, currH.Timestamp, epochDuration*msInS)
+	})
+}
+
+func reportHandler(c *cfg, l *zap.Logger) timer.Tick {
+	type report struct {
+		size, objsNum uint64
+	}
+	var (
+		m     sync.RWMutex
+		cache = make(map[cid.ID]report)
+	)
+
+	return func() {
+		epoch := c.CurrentEpoch()
 		st := c.cfgObject.cfgLocalStorage.localStorage
-		l := c.log.With(zap.String("component", "containerReports"), zap.Uint64("epoch", ev.EpochNumber()))
+		l = l.With(zap.Uint64("epoch", epoch))
```

Comment:
```suggestion
		l := l.With(zap.Uint64("epoch", epoch))
```

Reply:
does it help with smth?

Reply:
not clone the clone

https://go.dev/play/p/8_am7FVgVsW